crates/burn-jit/src/kernel/matmul/tune/base.rs

```diff
@@ -15,73 +16,45 @@ use crate::{
     out: JitTensor<R>,
-    _e: PhantomData<E>,
-}
-impl<R: JitRuntime, E: FloatElement> MatmulAutotuneOperationSet<R, E> {
-    fn new(lhs: JitTensor<R>, rhs: JitTensor<R>, out: JitTensor<R>) -> Self {
-        Self {
-            key: JitAutotuneKey::Matmul(MatmulAutotuneKey::new(&lhs.shape, &rhs.shape, E::dtype())),
-            lhs,
-            rhs,
-            out,
-            _e: PhantomData,
-        }
-    }
-}
+) {
+    let random_bounds: (E, E) = ((-10.0).elem::<E>(), (10.0).elem::<E>());
+    let lhs = random_like_uniform(lhs, random_bounds.0, random_bounds.1);
+    let rhs = random_like_uniform(rhs, random_bounds.0, random_bounds.1);
 
-impl<R: JitRuntime, E: FloatElement> AutotuneOperationSet<JitAutotuneKey>
-    for MatmulAutotuneOperationSet<R, E>
-{
-    fn key(&self) -> JitAutotuneKey {
-        self.key.clone()
-    }
-
-    fn autotunables(&self) -> Vec<Box<dyn AutotuneOperation>> {
-        let random_bounds: (E, E) = ((-10.0).elem::<E>(), (10.0).elem::<E>());
-        let lhs = random_like_uniform(&self.lhs, random_bounds.0, random_bounds.1);
-        let rhs = random_like_uniform(&self.rhs, random_bounds.0, random_bounds.1);
+    let out = empty_device::<R, E>(out.client.clone(), out.device.clone(), out.shape.clone());
 
-        let out = empty_device::<R, E>(
-            self.out.client.clone(),
-            self.out.device.clone(),
-            self.out.shape.clone(),
-        );
-
-        vec![
-            Box::new(MatmulTiling2d::<R, E>::new(
-                lhs.clone(),
-                rhs.clone(),
-                out.clone(),
-            )),
-            Box::new(MatmulAccelerated::<R, E>::new(
-                lhs.clone(),
-                rhs.clone(),
-                out.clone(),
-            )),
-            Box::new(MatmulSimple::<R, E>::new(
-                lhs.clone(),
-                rhs.clone(),
-                out.clone(),
-            )),
-        ]
-    }
+    tune_with!(lhs, rhs, out)
+}
 
-    fn fastest(self: Box<Self>, fastest_index: usize) -> Box<dyn AutotuneOperation> {
-        match fastest_index {
-            0 => Box::new(MatmulTiling2d::<R, E>::new(self.lhs, self.rhs, self.out)),
-            1 => Box::new(MatmulAccelerated::<R, E>::new(self.lhs, self.rhs, self.out)),
-            2 => Box::new(MatmulSimple::<R, E>::new(self.lhs, self.rhs, self.out)),
-            _ => panic!("Fastest index is out of bound"),
-        }
+fn should_run<R: JitRuntime, E: FloatElement>(
+    op: &MatmulOps<R, E>,
+    _key: &JitAutotuneKey,
+    index: usize,
+) -> bool {
+    match index {
+        // Accelerated
+        // TODO: Add way to query actual requirements from cubecl
+        1 => op.lhs.client.properties().feature_enabled(Feature::Cmma {
+            a: Elem::Float(FloatKind::F16),
+            b: Elem::Float(FloatKind::F16),
+            c: Elem::Float(FloatKind::F32),
+            m: 16,
+            k: 16,
+            n: 16,
+        }),
+        _ => true,
     }
 }
```

Comment:
We made all the checks in the matmul launch code. It's quite complex because it's not always the same instructions & precisions that are used! Normally, the function will panic on the checks, and therefore not be considered in the autotune. It will not run if the right features are not enabled on the device! `should_run` should not duplicate that logic.

Reply:
Which is why this came up to begin with